coalib/files/Fileproxy.py

```diff
@@ -0,0 +1,45 @@
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq("filename")
+class Fileproxy:
+    """
+    The file proxy is a wrapper object to wrap every file. The file proxy is an
+    object that contains properties about the file currently processed (such as
+    the filename). It contains:
+        * The filename, this is the starting point for every other attribute.
+        * The content of the file as a string.
+        * The content of the file, line-splitted.
+
+    Note: Other properties can be included later as and when required.
```

Comment:
I do not see the benefit of leaving a note like this in the API documentation.
Who is it helping?  API users?  Not likely.
Developers will do this, whether you give them permission or not.
When should this Note be removed?